client/state/posts/utils.js

```diff
@@ -241,6 +243,33 @@ export function normalizeTermsForApi( post ) {
 }
 
 /**
+ * Returns truthy if local terms object is different than API response
+ *
+ * @param  {Object}  localTermEdits local state of term edits
+ * @param  {Object}  savedTerms     term object returned from API POST
+ * @return {Boolean}                are there differences in local edits vs saved terms
+ */
+export function hasTermEditDifferences( localTermEdits, savedTerms ) {
+	if ( ! localTermEdits || ! savedTerms ) {
+		return false;
+	}
+
+	return !! find( localTermEdits, ( terms, taxonomy ) => {
```

Comment:
Hmm... how do we reconcile if there are more taxonomies in `savedTerms` than locally? Probably not something we'd expect to happen.